pages/index.js

```diff
@@ -68,17 +109,43 @@ export default function HomePage({ onAddEmotionEntry }) {
           On a scale from 0 to 100, how tense do you feel in this moment?
         </StyledLabel>
         <StyledInput
+          aria-label="Adjust tension level between 0 and 100"
           id="tension-level"
           name="tensionLevel"
           type="range"
           defaultValue={0}
           max={100}
+          onChange={(event) => setTension(event.target.value)}
         />
         <StyledWrapper>
           <StyledSpan>0</StyledSpan>
           <StyledSpan>100</StyledSpan>
         </StyledWrapper>
-        <StyledButton type="submit">Save</StyledButton>
+        {!isShown && <p>{tension}</p>}
+        {isShown && <StyledButton type="submit">Save</StyledButton>}
+
+        {!isShown && (
+          <StyledMessage>Your entry was successfully saved!</StyledMessage>
+        )}
+        {!isShown && (
+          <StyledButtonWrapper>
+            <StyledBackButton
+              type="reset"
+              value={"Done"}
+              onClick={() => {
+                setIsShown(!isShown);
+                setTension("0");
+              }}
+            ></StyledBackButton>
+            <StyledLink
+              $actionButton
+              href={{ pathname: "/create", query: { id: id } }}
+              forwardedAs={`/create`}
+            >
+              Add more details
+            </StyledLink>
+          </StyledButtonWrapper>
+        )}
```

Comment:
Line 124 should have the same state as line 125, to show the tension before submitting the form. Also you can improve readability (less complexity) by combining your UI rendering for each state:

```suggestion
        {!isFormSubmitted && (
          <>
            <p>{tension}</p>
            <StyledButton type="submit">Save</StyledButton>
          </>
        )}
        {isFormSubmitted && (
          <>
            <StyledButtonWrapper>
              <StyledBackButton
                type="reset"
                value={"Done"}
                onClick={() => {
                  setIsShown(!isShown);
                  setTension("0");
                }}
              ></StyledBackButton>
              <StyledLink
                $actionButton
                href={{ pathname: "/create", query: { id: id } }}
                forwardedAs={`/create`}
              >
                Add more details
              </StyledLink>
            </StyledButtonWrapper>
            <StyledMessage>Your entry was successfully saved!</StyledMessage>
          </>
        )}
```

Reply:
This should work exactly as you intend, or?

Reply:
Ah I forgot to change line 137

Reply:
Yes, thats works perfectly fine and is much more readable. Thanks again